marshal_test.go

```diff
@@ -188,16 +188,23 @@ func TestMarshalUnmarshal(t *testing.T) {
 				}
 			}
 
-			newType := reflect.New(reflect.TypeOf(type_))
-			err = unmarshal(&buf, newType.Interface())
-			if err != nil {
-				t.Fatal(fmt.Sprintf("%+v", err))
-				return
-			}
-
-			cmpType := reflect.Indirect(newType).Interface()
-			if !testEqual(type_, cmpType) {
-				t.Errorf("Roundtrip produced different results:\n %s", testDiff(type_, cmpType))
+			if type_ == nil {
+				err = unmarshal(&buf, nil)
+				if err != nil {
+					t.Fatal(fmt.Sprintf("%+v", err))
+					return
+				}
```

Comment:
Please put a return here and remove the `else`.

Reply:
Oops makes sense, sorry didn't spot that it was inside a func assumed it was a for loop